packages/user-operation-controller/src/types.ts

```diff
@@ -348,6 +349,17 @@ export type SwapsMetadata = {
   /** Symbol of the destination token. */
   destinationTokenSymbol: string | null;
 
+  approvalTxParams?: {
+    data: TransactionParams;
+    params?: {
+      requireApproval?: boolean;
+      type?: TransactionType;
+      swaps?: {
+        meta?: Partial<TransactionMeta>;
+      };
+    };
+  } | null;
+
```

Comment:
[nit] same comment as above, we can move this upwards to not break the alphabetical order. Preserving alphabetical order for better readability could be applied in other places of this PR as well